### PR TITLE
Micromed segments

### PR DESCRIPTION
**Pull request by samuelgarcia**
I though that micromed was mono segment but apparently it can be multi segment.
Here the implementation.

This include a test that need theses two files to be merge in gin  https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/pulls/143

### CODE REVIEW

**Comment by zm711** [comment]
@samuelgarcia do you think this is ready to go after getting the test files? Or do we want a milestone of 0.15.0 (for after next release).

**Comment by samuelgarcia** [comment]
I think this is ready to go.
Let me check again.

File: neo/rawio/micromedrawio.py [hunk 1]
```diff
@@ -99,25 +99,35 @@ def _parse_header(self):
...
-                "dtype": sig_dtype,
-                "order": "C",
-                "file_offset": 0,
-                "shape": signal_shape,
-            }
+
+            # "TRONCA" zone define segments
+            zname2, pos, length = zones["TRONCA"]
+            f.seek(pos)
+            max_segments = 100
```
Comment (zm711) [inline neo/rawio/micromedrawio.py:106]: where does this 100 come from. We probably need a comment for any magic number.

File: neo/rawio/micromedrawio.py [hunk 2]
```diff
@@ -99,25 +99,35 @@ def _parse_header(self):
...
+            self.info_segments = []
+            for i in range(max_segments):
+                seg_start = int(np.frombuffer(f.read(4), dtype="u4")[0])
```
Comment (zm711) [inline neo/rawio/micromedrawio.py:109]: Same here. Could we get a comment for why we are reading 4 at a time?

File: neo/rawio/micromedrawio.py [hunk 3]
```diff
@@ -155,6 +163,32 @@ def _parse_header(self):
...
+            self._buffer_descriptions = {0 :{}}
+            for seg_index in range(nb_segment):
+                seg_start, trace_offset = self.info_segments[seg_index]
+                self._t_starts.append(seg_start / self._sampling_rate)
+
+                start = seg_limits[seg_index]
+                stop = seg_limits[seg_index + 1]
+                
+                shape = (stop - start, Num_Chan)
+                file_offset = Data_Start_Offset + start * np.dtype(sig_dtype).itemsize * Num_Chan
```
Comment (zm711) [inline neo/rawio/micromedrawio.py:179]: any desire to add parentheses to help the reader with the math for this file_offset calculation?
```suggestion
                file_offset = Data_Start_Offset + ( start * np.dtype(sig_dtype).itemsize * Num_Chan)
```

File: neo/rawio/micromedrawio.py [hunk 4]
```diff
@@ -99,25 +99,35 @@ def _parse_header(self):
...
-                "type": "raw",
-                "file_path": str(self.filename),
-                "dtype": sig_dtype,
-                "order": "C",
-                "file_offset": 0,
-                "shape": signal_shape,
-            }
+
+            # "TRONCA" zone define segments
+            zname2, pos, length = zones["TRONCA"]
```
Comment (zm711) [inline neo/rawio/micromedrawio.py:104]: Light recommendation to have better name for `pos` :)

**Review by zm711 — COMMENTED** [review]
I think we should do a bit of a rewrite of this RawIO in general. Some of the variable naming is hard to follow and this low level seeking is also not super easy to follow. We should add some more comments to what is happening at the steps. But that can always be a future PR.

File: neo/rawio/micromedrawio.py (hunk 2 above)
Comment (samuelgarcia) [reply neo/rawio/micromedrawio.py:109]: 4 is 4bytes for "u4"

Comment (zm711) [reply neo/rawio/micromedrawio.py:111]: Sorry I meant why are you doing 'u4'? why not some other size for reading. How do you know reading u4 gets us what we need instead of some other size?  If we do a comment then we know that either 1) this size is documented or 2) you are doing this empirically.

File: neo/rawio/micromedrawio.py (hunk 4 above)
Comment (zm711) [reply neo/rawio/micromedrawio.py:104]: It's not crucial here :) But if any micromed issues come up I will send them your way.

**Comment by zm711** [comment]
To move forward with this I think you need to merge main into this branch to fix the CI/RTD builds. Then we can really see how the tests are doing :)

**Review by zm711 — APPROVED** [review]
